app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -0,0 +1,131 @@
+import { h, Component } from 'preact';
+import PropTypes from 'prop-types';
+
+import Navigation from './Navigation';
+import { getContentOfToken, updateOnboarding } from '../utilities';
+
+class ProfileForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+
+    this.state = {
+      summary: '',
+      location: '',
+      employment_title: '',
+      employer_name: '',
+      last_onboarding_page: 'personal info form', // i may need this i'm not sure exactly why yet
+    };
+  }
+
+  componentDidMount() {
+    updateOnboarding('bio form');
+  }
+
+  onSubmit() {
+    const csrfToken = getContentOfToken('csrf-token');
+    const { summary, location, employment_title, employer_name, last_onboarding_page } = this.state;
+    fetch('/onboarding_update', {
+      method: 'PATCH',
+      headers: {
+        'X-CSRF-Token': csrfToken,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user: { summary, location, employment_title, employer_name, last_onboarding_page } }),
+      credentials: 'same-origin',
+    }).then((response) => {
+      if (response.ok) {
+        const { next } = this.props;
+        next();
+      }
+    })
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  render() {
+    const { prev } = this.props;
+    return (
+      <div className="onboarding-main">
+        <Navigation prev={prev} next={this.onSubmit} />
+        <div className="onboarding-content about">
+          <header className="onboarding-content-header">
+            <h1 className="title">Build your profile</h1>
+            <h2 className="subtitle">
+              Tell us a little bit about yourself — this is how others will see
+              you on DEV. You’ll always be able to edit this later in your
+              Settings.
+            </h2>
+          </header>
+          <div className="current-user-info">
+            <figure className="current-user-avatar-container">
+              <img className="current-user-avatar" alt="profile" src={window.userData().profile_image_90} />
+            </figure>
+            <h3>{window.userData().name}</h3>
+            <p>{window.userData().username}</p>
```

Comment:
Feels weird to me to call for a function on `window` directly in the jsx here...I'd say we should extract it and set it on the component. Also, what happens if the `userData()` isn't loaded and this tries to render? Can we do this earlier on in the constructor or when the component mounts?

Reply:
Updated!